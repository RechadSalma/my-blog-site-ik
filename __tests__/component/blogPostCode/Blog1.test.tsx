import { render } from "@testing-library/react";
import Blog1Codes from "../../../components/blogPostCodes/Blog1";

describe("<Blog1/> code for blog1 component", () => {
  const blogOneComponent = () => render(<Blog1Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogOneComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog1 code tag", () => {
    const { container } = blogOneComponent();

    expect(container).toMatchSnapshot();
  });
});
