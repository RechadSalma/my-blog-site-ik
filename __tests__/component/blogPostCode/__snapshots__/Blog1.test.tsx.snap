// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog1/> code for blog1 component snapshot Blog1 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div>
        <span
          class="keyword"
        >
          function
        </span>
          createTimestampInSqlFormat () {
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         date = 
        <span
          class="keyword"
        >
          new
        </span>
         
        <span
          class="method"
        >
          Date
        </span>
        ();
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         year =  date
        <span
          class="method"
        >
          .getFullYear
        </span>
        ();
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         month =  date
        <span
          class="method"
        >
          .getMonth
        </span>
        () + 
        <span
          class="value"
        >
          1
        </span>
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         day =  date
        <span
          class="method"
        >
          .getDate
        </span>
        ();
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         hour =  date
        <span
          class="method"
        >
          .getHours
        </span>
        ();
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         minute =  date
        <span
          class="method"
        >
          .getMinutes
        </span>
        ();
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         second =  date
        <span
          class="method"
        >
          .getSeconds
        </span>
        ();
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const
        </span>
         timestampArray = [year, month, day, hour, minute, seconds];
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          let
        </span>
         timestampString = ""
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          for
        </span>
         (
        <span
          class="keyword"
        >
          let
        </span>
         dateNumber 
        <span
          class="keyword"
        >
          of
        </span>
          timestampArray) {
      </div>
      <div
        class="comment tab2"
      >
        // if number is less than 10 than add a 0 in front of it
      </div>
      <div
        class="tab2"
      >
        <span
          class="keyword"
        >
          if
        </span>
         (dateNumber &gt;= 
        <span
          class="value"
        >
          10
        </span>
        ) {
      </div>
      <div
        class="tab3"
      >
        timestampString += dateNumber;
      </div>
      <div
        class="tab2"
      >
        } 
        <span
          class="keyword"
        >
          else
        </span>
         {
      </div>
      <div
        class="tab3"
      >
        timestamString += \`
        <span
          class="value"
        >
          0
        </span>
        \${dateNumber}\`;
      </div>
      <div
        class="tab2"
      >
        }
      </div>
      <div
        class="tab1"
      >
        }
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const/
        </span>
         timestampRegEx = /(\\d{4})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})/g;
      </div>
      <br />
      <div
        class="comment tab1"
      >
        // create the sql timestamp format.You could change the format here if need be 
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          const/
        </span>
         timestamp = timestampString
        <span
          class="method"
        >
          .replace
        </span>
        (
      </div>
      <div
        class="tab2"
      >
        timestampRegEx,
      </div>
      <div
        class="tab2"
      >
        <span
          class="value"
        >
          "$1-$2-$3 $4:$5:$6"
        </span>
      </div>
      <div
        class="tab1"
      >
        );
      </div>
      <br />
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          return
        </span>
         timestamp;
      </div>
      <div>
        }
      </div>
    </code>
  </pre>
</div>
`;
